src/modules/fundingManager/oracle/FM_PC_ExternalPrice_Redeeming_v1.sol

```diff
@@ -0,0 +1,698 @@
+ *              - IFM_PC_ExternalPrice_Redeeming_v1
+ *              - ERC20PaymentClientBase_v1
+ *              - RedeemingBondingCurveBase_v1
+ *          Key features:
+ *              - External price integration
+ *              - Payment client functionality
+ *          Price feeds are managed through an external oracle contract that
+ *          implements IOraclePrice_v1.
+ *
+ * @custom:security-contact security@inverter.network
+ *                          In case of any concerns or findings, please refer to
+ *                          our Security Policy at security.inverter.network or
+ *                          email us directly!
+ *
+ * @custom:version   1.0.0
+ *
+ * @custom:standard-version  1.0.0
+ *
+ * @author  Zealynx Security
+ */
+contract FM_PC_ExternalPrice_Redeeming_v1 is
+    IFM_PC_ExternalPrice_Redeeming_v1,
+    ERC20PaymentClientBase_v1,
+    RedeemingBondingCurveBase_v1
+{
+    /// @inheritdoc ERC165Upgradeable
+    /// @param interfaceId_ The interface identifier to check support for
+    /// @return True if the interface is supported
+    function supportsInterface(bytes4 interfaceId_)
+        public
+        view
+        override(ERC20PaymentClientBase_v1, RedeemingBondingCurveBase_v1)
+        returns (bool)
+    {
+        return interfaceId_
+            == type(IFM_PC_ExternalPrice_Redeeming_v1).interfaceId
+            || super.supportsInterface(interfaceId_);
+    }
+
+    using SafeERC20 for IERC20;
+
+    //--------------------------------------------------------------------------
+    // Constants
+
+    /// @dev Value is used to convert deposit amount to 18 decimals
+    uint8 private constant EIGHTEEN_DECIMALS = 18;
+
+    /// @notice Role for whitelisted addresses
+    bytes32 public constant WHITELIST_ROLE = "WHITELIST_ROLE";
+
+    /// @notice Role for payment queue
+    bytes32 public constant QUEUE_MANAGER_ROLE = "QUEUE_MANAGER_ROLE";
+
+    //--------------------------------------------------------------------------
+    // State Variables
+
+    /// @dev Oracle price feed contract used for price discovery
+    /// @notice Contract that provides external price information for token valuation
+    IOraclePrice_v1 private _oracle;
+
+    /// @dev Token that is accepted by this funding manager for deposits
+    /// @notice The ERC20 token contract used for collateral in this funding manager
+    IERC20 private _token;
+
+    /// @dev Token decimals of the issuance token
+    /// @notice Number of decimal places used by the issuance token for proper decimal handling
+    uint8 private _issuanceTokenDecimals;
+
+    /// @dev Token decimals of the Orchestrator token
+    /// @notice Number of decimal places used by the collateral token for proper decimal handling
+    uint8 private _collateralTokenDecimals;
+
+    /// @dev Maximum fee that can be charged for sell operations, in basis points
+    /// @notice Maximum allowed project fee percentage for selling tokens
+    uint private _maxSellFee;
```

Comment:
General question: I find it a bit confusing to just call it "sellFee", as that could either mean the project or collateral fee - or both. If this is just about the project fee, I'd rename this to clearly indicate this.